BGL/include/CGAL/boost/graph/helpers.h

```diff
@@ -960,24 +941,83 @@ void swap_edges(const typename boost::graph_traits<FaceGraph>::halfedge_descript
  * \ingroup PkgBGLHelperFct
  *
  * removes all vertices, faces and halfedges from a graph. Calls
- * `remove_edge()`, `remove_vertex()`, and `remove_face()` for each
- * edge, vertex or face.
+ * \link MutableHalfedgeGraph `remove_vertex()`\endlink,
+ * \link MutableHalfedgeGraph `remove_edge()`\endlink, and
+ * \link MutableFaceGraph `remove_face()`\endlink, for each vertex, edge, and face.
+ *
+ * @warning This function does not perform anything more than what is advertised above. It is
+ * up to the user to e.g. clean garbage or remove internal property maps (if relevant, and desired).
+ *
+ * @tparam FaceGraph model of `MutableHalfedgeGraph` and `MutableFaceGraph`
+ *
+ * @param g the graph whose elements will be removed
+ *
+ * @sa `CGAL::clear()`
+ **/
+template<typename FaceGraph>
+void remove_all_elements(FaceGraph& g)
```

Comment:
I would update `Surface_mesh` package to have its own overload.

Reply:
Polyhedron could also.

Reply:
> I would update `Surface_mesh` package to have its own overload.

Like [146f9ec](https://github.com/CGAL/cgal/pull/7439/commits/146f9ecb21db4815354058f6e061ccffaf93df9d)? Not sure that I like specializing BGL functions that aren't trivial, it defeats the point of the common API.

Reply:
Yes. I get your point but for dynamic property map we also have something taking advantage of the index based of surface mesh so it is somehow similar. In both cases it is done for efficiency reasons.